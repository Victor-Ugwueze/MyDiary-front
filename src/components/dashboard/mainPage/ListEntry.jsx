// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const ListEntrySection = ({ current, openNewEntryModal }) => (
  <div id='diary' className={`dashboard-section tab-pane ${current && 'selected'} container`}>
    <div id='add-entry'>
      <button className='btn btn-primary add-entry' data-target='add-new-entry' onClick={openNewEntryModal}>
        New Entry
      </button>
    </div>
    <div id='dairy-content'>
      <div id='diary-heading'>
        <h4 id='title'>Diary Entry</h4>
        <h4 className='entry-head' id='created-at'>
          Date Cretated
        </h4>
        <h4 className='entry-head' id='action'>
          Action
        </h4>
      </div>
      <ul id='dairy-entries' />
    </div>
    <div className='pagination' />
    <input type='hidden' id='entryCount' />
  </div>
);

ListEntrySection.propTypes = {
  current: PropTypes.bool,
  openNewEntryModal: PropTypes.func,
};

export default ListEntrySection;
