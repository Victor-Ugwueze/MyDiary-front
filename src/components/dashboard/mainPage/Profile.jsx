// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const ProfileSection = ({ current }) => (
  <div id='profile' className={`profile-section ${current && 'selected'} tab-pane`}>
    <div className='user-profile'>
      {/* <div id="account">
       <div className="picture">
           <!-- <img src="/images/profile.jpg" id="profile-picture" />
           <div id="upload-input-icon">
               <img src="/images/icons/add_photo_white.png" />
               <input type="file" className="profile-upload" accept="image/*" />
         </div>
       </div>
   </div> */}
      <div className='profile-details'>
        <div>
          <form encType='application/x-www-htmlform-urlencoded' id='update-profile'>
            <h4>Profile Details</h4>
            <div className='errors' id='profile-error' />
            <div className='item'>
              <label htmlFor='firstname'>First Name</label>
              <input type='text' className='htmlform-input' name='firstName' />
            </div>
            <div className='item'>
              <label htmlFor='lastname'>Last Name</label>
              <input type='text' className='htmlform-input' name='lastName' />
            </div>
            <div className='item'>
              <label htmlFor='email'>Email</label>
              <input type='email' className='htmlform-input' name='email' />
            </div>
            <div className='item'>
              <label htmlFor='location'>Location</label>
              <input id='location' type='text' className='htmlform-input' name='location' />
            </div>
            <button
              data-target='update-password'
              type='button'
              className='btn btn-primary change-password'>
              Change password
            </button>
            <button id='save' type='submit' className='btn btn-primary' disabled>
              Save
            </button>
            <span className='hide saving-settings-indicator'>Saving...</span>
          </form>
        </div>
        <div id='total-entry-created'>
          <h4>Activity</h4>
          <span id='entries_created' />
          <div>
            <p className='joined date'>
              Joined <span id='date-registered' />
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
);

ProfileSection.propTypes = {
  current: PropTypes.bool,
};

export default ProfileSection;
