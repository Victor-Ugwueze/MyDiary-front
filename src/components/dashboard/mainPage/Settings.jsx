// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

const SettingsSection = ({ current }) => (
  <div id='settings' className={`dashboard-section tab-pane ${current && 'selected'}`}>
    <div className='setting-page row'>
      <ul>
        <li>
          <h4 id='setting-title'>Notifications</h4>
          <div className='Notification-list'>
            <div id='add-notification-item'>{/* <a href="">+ Add new reminnder</a> */}</div>
            <div className='Notification-item'>
              <p className='title'>We will remind you to write your journal</p>
              <input type='checkbox' id='reminder-journal' className='email-subscribe-update' />
              <span className='hide saving-settings-indicator'>Saving...</span>
            </div>
          </div>
        </li>
        <li>
          <h4>News Letter</h4>
          <div id='news-letter'>
            <p className='title'>Receive monthly news Letter</p>
            <input type='checkbox' id='receive-newsletter' className='email-subscribe-update' />
            <span className='hide saving-settings-indicator'>Saving...</span>
          </div>
        </li>
      </ul>
    </div>
  </div>
);

SettingsSection.propTypes = {
  current: PropTypes.bool,
};

export default SettingsSection;
