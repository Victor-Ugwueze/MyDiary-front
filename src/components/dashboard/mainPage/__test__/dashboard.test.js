// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// components
import ListEntry from '../ListEntry';
import Settings from '../Settings';
import Profile from '../Profile';


configure({
  adapter: new Adapter(),
});

describe('Renders modals', () => {
  it('should render the list section of dashboard page', () => {
    const component = shallow(
      <ListEntry current={true}/>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the settings section of dashboard page', () => {
    const component = shallow(
      <Settings current={false}/>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the profile section of dashboard page', () => {
    const component = shallow(
      <Profile current={false}/>
    );
    expect(component).toMatchSnapshot();
  });
});
