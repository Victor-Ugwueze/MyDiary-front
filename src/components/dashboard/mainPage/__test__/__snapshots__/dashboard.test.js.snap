// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders modals should render the list section of dashboard page 1`] = `
<div
  className="dashboard-section tab-pane selected container"
  id="diary"
>
  <div
    id="add-entry"
  >
    <button
      className="btn btn-primary add-entry"
      data-target="add-new-entry"
    >
      New Entry
    </button>
  </div>
  <div
    id="dairy-content"
  >
    <div
      id="diary-heading"
    >
      <h4
        id="title"
      >
        Diary Entry
      </h4>
      <h4
        className="entry-head"
        id="created-at"
      >
        Date Cretated
      </h4>
      <h4
        className="entry-head"
        id="action"
      >
        Action
      </h4>
    </div>
    <ul
      id="dairy-entries"
    />
  </div>
  <div
    className="pagination"
  />
  <input
    id="entryCount"
    type="hidden"
  />
</div>
`;

exports[`Renders modals should render the profile section of dashboard page 1`] = `
<div
  className="profile-section false tab-pane"
  id="profile"
>
  <div
    className="user-profile"
  >
    <div
      className="profile-details"
    >
      <div>
        <form
          encType="application/x-www-htmlform-urlencoded"
          id="update-profile"
        >
          <h4>
            Profile Details
          </h4>
          <div
            className="errors"
            id="profile-error"
          />
          <div
            className="item"
          >
            <label
              htmlFor="firstname"
            >
              First Name
            </label>
            <input
              className="htmlform-input"
              name="firstName"
              type="text"
            />
          </div>
          <div
            className="item"
          >
            <label
              htmlFor="lastname"
            >
              Last Name
            </label>
            <input
              className="htmlform-input"
              name="lastName"
              type="text"
            />
          </div>
          <div
            className="item"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="htmlform-input"
              name="email"
              type="email"
            />
          </div>
          <div
            className="item"
          >
            <label
              htmlFor="location"
            >
              Location
            </label>
            <input
              className="htmlform-input"
              id="location"
              name="location"
              type="text"
            />
          </div>
          <button
            className="btn btn-primary change-password"
            data-target="update-password"
            type="button"
          >
            Change password
          </button>
          <button
            className="btn btn-primary"
            disabled={true}
            id="save"
            type="submit"
          >
            Save
          </button>
          <span
            className="hide saving-settings-indicator"
          >
            Saving...
          </span>
        </form>
      </div>
      <div
        id="total-entry-created"
      >
        <h4>
          Activity
        </h4>
        <span
          id="entries_created"
        />
        <div>
          <p
            className="joined date"
          >
            Joined 
            <span
              id="date-registered"
            />
          </p>
        </div>
      </div>
    </div>
  </div>
  <ChangePasswordModal />
</div>
`;

exports[`Renders modals should render the settings section of dashboard page 1`] = `
<div
  className="dashboard-section tab-pane false"
  id="settings"
>
  <div
    className="setting-page row"
  >
    <ul>
      <li>
        <h4
          id="setting-title"
        >
          Notifications
        </h4>
        <div
          className="Notification-list"
        >
          <div
            id="add-notification-item"
          />
          <div
            className="Notification-item"
          >
            <p
              className="title"
            >
              We will remind you to write your journal
            </p>
            <input
              className="email-subscribe-update"
              id="reminder-journal"
              type="checkbox"
            />
            <span
              className="hide saving-settings-indicator"
            >
              Saving...
            </span>
          </div>
        </div>
      </li>
      <li>
        <h4>
          News Letter
        </h4>
        <div
          id="news-letter"
        >
          <p
            className="title"
          >
            Receive monthly news Letter
          </p>
          <input
            className="email-subscribe-update"
            id="receive-newsletter"
            type="checkbox"
          />
          <span
            className="hide saving-settings-indicator"
          >
            Saving...
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
